### PR TITLE
Arcadia build fixes

### PR DESCRIPTION
**Pull request by proller**

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
@abyss7 Waiting for your review.

**Comment by alexey-milovidov** [comment]
@abyss7 ?

**Comment by abyss7** [comment]
> @abyss7 Waiting for your review.

I have viewed the changes, but can't reliably check their purpose - there are no any build logs from `yamake`: link to the internal review would be appreciated. At least the CI builds should be restarted to make sure that release-branch isn't broken.

**Comment by proller** [comment]
https://a.yandex-team.ru/review/912102

**Comment by alexey-milovidov** [comment]
@proller Please merge with master - because no tests have run.

**Comment by abyss7** [comment]
> https://a.yandex-team.ru/review/912102

There are still 2 build errors in the 8th iteration.

**Comment by proller** [comment]
because macos build broken and one insternal problem not solved

**Comment by abyss7** [comment]
Are all those fixes in the master already? Otherwise we'll have make them again in the next port to arcadia.

File: CMakeLists.txt [hunk 1]
```diff
@@ -255,10 +255,10 @@ if (USE_STATIC_LIBRARIES AND HAVE_NO_PIE)
     set (CMAKE_C_FLAGS                   "${CMAKE_C_FLAGS} ${FLAG_NO_PIE}")
 endif ()
 
-if (NOT SANITIZE)
+if (NOT SANITIZE AND NOT SPLIT_SHARED_LIBRARIES)
```
Comment (alexey-milovidov) [inline CMakeLists.txt]: This is most questionable part. I will remove.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by abyss7** [comment]
@alexey-milovidov Have you checked that all the made changes are necessary in the way they are made? Why to remove me as a reviewer?

**Comment by alexey-milovidov** [comment]
> Have you checked that all the made changes are necessary in the way they are made?

I've read all the changes and noticed that some changes are already in master and some changes have their motivation.

> Why to remove me as a reviewer?

Don't want to wait.

**Comment by alexey-milovidov** [comment]
Now I'm waiting for build check. Then I will merge and remove MiAllocator.

**Pull request merged**